src/super_gradients/training/datasets/detection_datasets/coco_format_detection.py

```diff
@@ -124,7 +147,9 @@ def _load_annotation(self, sample_id: int) -> dict:
         :return img_path:               Path to the associated image
         """
 
-        annotation = self._annotations[sample_id]
+        start_addr = 0 if sample_id == 0 else self._addr[sample_id - 1].item()
+        end_addr = self._addr[sample_id].item()
+        annotation = pickle.loads(self._annotations[start_addr:end_addr].numpy().data)
```

Comment:
We saw slowdown due to pickle load, do we want to make the serialize-parse fix optional?
I mean, eventually a memory leak is  a memory leak, but for small datasets you get overhead whereas without the fix "you'd be fine". @BloodAxe , thoughts?

Reply:
@NatanBagrov yes IMO